### ISSUE TITLE
About SAC actor_loss queries. I dont understand the loss is postive.

### ISSUE THREAD

**Issue by www2171668** [HorizonRobotics/alf#1204]
Helloe. I see SAC code. In _actor_train_step(), it calculates actor_loss by following code. There are two things I don't understand.
1、why this actor_loss is positive? By element_wise_squared_loss, the dqda is positive. But in sac paper, the Q_value is negative and we need to minimize -Q. 
2、whty we need to calculate losses.element_wise_squared_loss((dqda + action).detach(), action)? It seems get dqda^2. I dont know why we're squared dq/da.

`        dqda = nest_utils.grad(action, q_value.sum())

        def actor_loss_fn(dqda, action):
            loss = 0.5 * losses.element_wise_squared_loss((dqda + action).detach(), action)
            return loss.sum(list(range(1, loss.ndim)))

        actor_loss = nest.map_structure(actor_loss_fn, dqda, action)
        actor_loss = math_ops.add_n(nest.flatten(actor_loss))
        actor_info = LossInfo(
            loss=actor_loss + cont_alpha * continuous_log_pi,
            extra=SacActorInfo(actor_loss=actor_loss, neg_entropy=neg_entropy))`
Thankyou.

**Comment by Haichao-Zhang** [comment]
Hi Thanks for the question. In short, 
1. The actor loss in ALF has a different formulation but has an equivalent gradient to the original loss in the paper in terms of minimization. We used this loss since it is more interpretable as it is non-negative (as you noticed), in contrast to the original loss, which can have an unbounded range.
2. This is our specially construct loss. The derivations below show the equivalence of its gradient with the one in the paper (illustrated with a single sample, and omit the entropy part since that part is identical with SAC paper):

The loss in ALF： 
<img width="357" alt="image" src="https://user-images.githubusercontent.com/21375027/154525687-78f1caec-6665-44ac-b884-41bbbe1a3dbd.png">

Its gradient wrt NN parameters:
<img width="413" alt="image" src="https://user-images.githubusercontent.com/21375027/154527595-c2a570c8-d9bc-4d5a-8fda-24a3b45efbe7.png">


As you can see, this is equivalent to the gradient for maximizing Q.

Hope this can address your questions on the actor loss in ALF and let us know if you have any further questions. Thanks.

**Comment by www2171668** [comment]
Your answer is clear and detailed. It's really useful and also helps me to understand other algorithms like DDPG . Thank you very much.
